viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -32,30 +27,22 @@ struct ColorARGB {
     unsigned char B;
 };
 
-// Check if pixel is the default color signaling no data is present
-bool CheckIfEmptyPixel(ColorARGB pixel_color) {
-    return (pixel_color.R == defaultNoDataRGBValue) &&
-           (pixel_color.G == defaultNoDataRGBValue) &&
-           (pixel_color.B == defaultNoDataRGBValue);
-}
+// Check the green color channel to filter unobserved pixels
+bool CheckIfEmptyPixel(ColorARGB pixel_color) { return (pixel_color.G == 0); }
 
-// Convert the scale of a specified pixel color channel from the given
-// 102-255 range to 100-0. This is an initial solution for extracting
-// probability information from cartographer
+// Convert the scale of a specified color channel from the given UCHAR 
+// range of 0 - 255 to an inverse probability range of 100 - 0
 int CalculateProbabilityFromColorChannels(ColorARGB pixel_color) {
     unsigned char max_val = UCHAR_MAX;
-    unsigned char min_val = defaultNoDataRGBValue;
+    unsigned char min_val = 0;
     unsigned char max_prob = 100;
     unsigned char min_prob = 0;
 
-    // Probability is currently determined solely by the R channel
-    // Values are restricted to be above or equal to defaultNoDataRGBValue
-    // to ensure negative probabilities cannot occur
-    // https://viam.atlassian.net/browse/RSDK-2567
+    // Probability is currently determined solely by the red color channel
```

Comment:
Can we please have in the comment the fact that the higher the red channel value, the lower the probability (I only realized this recently)?

Reply:
the "inverse probability" was intended to cover this I believe but I do agree it could be clearer